dotnet/src/SemanticKernel.UnitTests/AI/ChatCompletion/ChatHistoryTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.AI.ChatCompletion;
+using Xunit;
+
+namespace SemanticKernel.UnitTests.AI.ChatCompletion;
+
+/// <summary>
+/// Unit tests of <see cref="ChatHistory"/>.
+/// </summary>
+public class ChatHistoryTests
+{
+    [Fact]
+    public void ItCanBeSerialised()
+    {
+        // Arrange
+        var options = new JsonSerializerOptions();
+        options.Converters.Add(new AuthorRoleConverter());
```

Comment:
Should the AuthorRole property instead be tagged with [JsonConverter(typeof(AuthorRoleConverter))] so that this isn't necessary?

Reply:
I can use `[JsonConstructor]` instead